code/game/machinery/computer/robot.dm

```diff
@@ -167,6 +167,8 @@
 				s.start()
 				D.visible_message("<span class='danger'>\the [D] self destructs!</span>")
 				D.gib()
+				message_admins("[ADMIN_LOOKUPFLW(usr)] detonated [key_name_admin(D)][ADMIN_JMP(D)]!")

```

Comment:
add a `get_turf(D)` before gibbing and do `ADMIN_JMP` on that turf, also send the messages before gib is called to avoid any possible problem with `D` no longer existing.